### PR TITLE
[CELEBORN-1211] Add extension for celeborn shuffle handler

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
1. Add extension API to CelebornShuffleHandler.

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
GA.

### CODE REVIEW

File: client/src/main/java/org/apache/celeborn/client/ShuffleClient.java [hunk 1]
```diff
@@ -142,13 +130,12 @@ public static void printReadStats(Logger logger) {
...
   }
 
-  public void setupIoCrypto(
-      Optional<byte[]> ioCryptoKey, Properties ioCryptoConf, byte[] ioCryptoInitializationVector) {}
-
   public abstract void setupLifecycleManagerRef(String host, int port);
 
   public abstract void setupLifecycleManagerRef(RpcEndpointRef endpointRef);
 
+  public abstract void setupExtension(byte[] extension);
```
Comment (waitinfuture) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClient.java:137]: Better to use `setExtension`

**Review by waitinfuture — APPROVED** [review]
LGTM except a nit, cc @pan3793 @RexXiong

**Comment by pan3793** [comment]
would the change be clearer if we do a clean revert first?

**Comment by waitinfuture** [comment]
> would the change be clearer if we do a clean revert first?

I think you are right. I suggested not to revert yesterday, but now I think revert is clearer.

File: client/src/main/java/org/apache/celeborn/client/ShuffleClient.java [hunk 2]
```diff
@@ -122,6 +134,8 @@ public static void printReadStats(Logger logger) {
 
   public abstract void setupLifecycleManagerRef(RpcEndpointRef endpointRef);
 
+  public abstract void setExtension(byte[] extension);
```
Comment (pan3793) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClient.java:137]: can we have docs for that?

**Comment by mridulm** [comment]
Can you provide more context about the legal reasons please ?
I did not find any details in the jira CELEBORN-1211 as well.

**Comment by waitinfuture** [comment]
> Can you provide more context about the legal reasons please ? I did not find any details in the jira [CELEBORN-1211](https://issues.apache.org/jira/browse/CELEBORN-1211) as well.

Hi @mridulm , we found instructions about adding cryptography features in this doc:
https://infra.apache.org/crypto.html

image

and we decided not to include such code for now.

**Comment by mridulm** [comment]
Sounds good, btw going through the process should not be very cumbersome - we can get help from folks who have done it in past at Apache projects :-)

**Comment by waitinfuture** [comment]
> Sounds good, btw going through the process should not be very cumbersome - we can get help from folks who have done it in past at Apache projects :-)

That will be great. The reverted crypto logic was only for our internal use,  if it's required by community in the future, we can discuss about the process :)

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main(v0.5.0)/branch-0.4(v0.4.0)

**Pull request closed without merging**